### PR TITLE
Add A CITATION.cff File

### PR DESCRIPTION
**Pull request by astroesteban**
| | |
|:---|:---|
|**_Originating Project/Creator_**|@astroesteban |
|**_Affected Component_**| Repo Main Page |
|**_Affected Architectures(s)_**| None |
|**_Related Issue(s)_**| #946 |
|**_Has Unit Tests (y/n)_**| n |
|**_Builds Without Errors (y/n)_**| y |
|**_Unit Tests Pass (y/n)_**|  |
|**_Documentation Included (y/n)_**|  |

---
## Change Description

Add a _CITATION.cff_ file to the repo easily citing this repository in APA or BibTeX format

## Rationale

GitHub recently announced [support for citations](https://github.blog/2021-08-19-enhanced-support-citations-github/) using  `ruby-cff`. In order to add a citation for F' we just need to create a `CITATION.cff` file.

By adding this file we make it easier for users to cite F' thus bolstering the recognition of the project and boosting the reference count on sites like Google Scholar. This could have potential benefit when making a business case for F', recognizes the work made by contributors, and making it easier for people to reference the project is just nice.

## Testing/Review Recommendations

I added the authors listed [on this paper](https://digitalcommons.usu.edu/cgi/viewcontent.cgi?article=4140&context=smallsat). This section is up for debate. We could:

1. Add all the current contributors and keep updating this file as more contributors join the project.
2. Keep the current author list as is based on the paper.
3. Reduce the list of authors to the minimum of at least one. 

## Future Work

Note any additional work that will be done relating to this issue.

### CODE REVIEW

**Comment by LeStarch** [comment]
Should have responded here:  we do like the idea and our soliciting a set of names to add above the ones you have already added.

Thanks for pointing this out!

**Comment by ThibFrgsGmz** [comment]
It may also be worthwhile to include a "Citation" section at the end of the project's README file.

This is something I've noticed on the Facebook Research Unit projects (https://github.com/facebookresearch), like:

- MMF: https://github.com/facebookresearch/mmf
- Detectron2: https://github.com/facebookresearch/detectron2

**Comment by astroesteban** [comment]
@ThibFrgsGmz I _believe_ this GitHub feature supplants the need for having the citation in the README itself. Having GitHub render the citation for you via the CITATION.cff is really just a convenience feature.

**Comment by ThibFrgsGmz** [comment]
@astroesteban Indeed, now that you say it, I realize it.
It will be useless with .cff, and it will clutter up the README, which seems to me to be quite overloaded.

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+- family-names: "Canham"
+  given-names: "Timothy"
+- family-names: "Watney"
+  given-names: "Garth"
+- family-names: "Reder"
+  given-names: "Leonard"
+- family-names: "Levison"
+  given-names: "Jeffrey"
+title: "F': A Flight-Proven, Multi-Platform, Open-Source Flight Software Framework"
+url: "https://github.com/nasa/fprime"
```
Comment (jsoref) [inline CITATION.cff:15]: Could I encourage you to include a newline at the end of the file (that would get rid of the strange ⛔ thing and mean that the next person to touch the file w/ various tools won't be blamed for touching the last line in the file.

**Comment by LeStarch** [comment]
@astroesteban I asked around about this, and figured out what the hesitation is: authors.  We'd rather not publish formal authors as part of this file as it would fail to properly capture the full set of contributors. We'd prefer to just publish the url/message/title lines.  

Would you like to make that adjustment and move this forward, or would you like me to do so?

**Comment by astroesteban** [comment]
Totally agree @LeStarch. I can make that change later today 👍

**Review by LeStarch — APPROVED** [review]

**Pull request merged**